Sources/BraveWallet/Panels/Signature Request/SignInWithEthereumView.swift

```diff
@@ -0,0 +1,243 @@
+  
+  @State private var isShowingDetails: Bool = false
+  @Environment(\.sizeCategory) private var sizeCategory
+  
+  var body: some View {
+    ScrollView {
+      VStack(spacing: 10) {
+        faviconAndOrigin
+        
+        messageContainer
+        
+        buttonsContainer
+          .padding(.top)
+          .opacity(sizeCategory.isAccessibilityCategory ? 0 : 1)
+          .accessibility(hidden: sizeCategory.isAccessibilityCategory)
+      }
+      .padding()
+    }
+    .overlay(
+      Group {
+        if sizeCategory.isAccessibilityCategory {
+          buttonsContainer
+            .frame(maxWidth: .infinity)
+            .padding(.top)
+            .background(
+              LinearGradient(
+                stops: [
+                  .init(color: Color(.braveGroupedBackground).opacity(0), location: 0),
+                  .init(color: Color(.braveGroupedBackground).opacity(1), location: 0.05),
+                  .init(color: Color(.braveGroupedBackground).opacity(1), location: 1),
+                ],
+                startPoint: .top,
+                endPoint: .bottom
+              )
+              .ignoresSafeArea()
+              .allowsHitTesting(false)
+            )
+        }
+      },
+      alignment: .bottom
+    )
+    .background(Color(braveSystemName: .containerHighlight))
+    .navigationTitle(Strings.Wallet.signInWithBraveWallet)
+  }
+  
+  private var faviconAndOrigin: some View {
+    VStack(spacing: 8) {
+      OriginInfoFavicon(originInfo: originInfo)
+      Text(verbatim: originInfo.eTldPlusOne)
+      Text(originInfo: originInfo)
+        .font(.caption)
+        .foregroundColor(Color(.braveLabel))
+        .multilineTextAlignment(.center)
+    }
+  }
+  
+  private var messageContainer: some View {
+    VStack(alignment: .leading, spacing: 10) {
+      AddressView(address: account.address) {
+        AccountView(
+          address: account.address,
+          name: account.name,
+          blockieShape: .rectangle
+        )
+      }
+      
+      // 'You are signing into xyz. Brave Wallet will share your wallet address with xyz.'
+      Text(String.localizedStringWithFormat(
+        Strings.Wallet.signInWithBraveWalletMessage,
+        originInfo.eTldPlusOne, originInfo.eTldPlusOne
+      ))
+      
+      NavigationLink(
+        destination: SignInWithEthereumDetailsView(
+          originInfo: originInfo,
+          message: message
+        )
+      ) {
+        Text(Strings.Wallet.seeDetailsButtonTitle)
+          .fontWeight(.semibold)
+          .foregroundColor(Color(braveSystemName: .textInteractive))
+          .contentShape(Rectangle())
+      }
+      
+      if let statement = message.statement, let resources = message.resources {
+        Divider()
+        
+        VStack(alignment: .leading, spacing: 6) {
+          Text(Strings.Wallet.siweMessageLabel)
+            .fontWeight(.semibold)
+          Text(verbatim: statement)
+            .textSelection(.enabled)
```

Comment:
From a design perspective maybe we want to use `headine`/`subheadline` font styles for these types of header/detail text? (Same for below)